From 6d5a941f6212fe5b5c725c86d10d9b0be0f151cb Mon Sep 17 00:00:00 2001
From: Ghislain Vaillant <ghisvail@gmail.com>
Date: Sat, 11 Feb 2023 20:06:38 +0100
Subject: [PATCH] Fix mouse issues on Wayland


diff --git a/src/video/sdl2_v.cpp b/src/video/sdl2_v.cpp
index 51c490d23..c7870a8f2 100644
--- a/src/video/sdl2_v.cpp
+++ b/src/video/sdl2_v.cpp
@@ -370,10 +370,17 @@ bool VideoDriver_SDL_Base::PollEvent()
 
 	if (!SDL_PollEvent(&ev)) return false;
 
+	const char *sdl_video_driver = SDL_GetCurrentVideoDriver();
+	const bool sdl_on_wayland =
+		(sdl_video_driver != nullptr) ? std::string(sdl_video_driver) == "wayland" : false;
+
 	switch (ev.type) {
 		case SDL_MOUSEMOTION:
-			if (_cursor.UpdateCursorPosition(ev.motion.x, ev.motion.y, true)) {
+			if (_cursor.UpdateCursorPosition(ev.motion.x, ev.motion.y, !sdl_on_wayland)) {
+				SDL_SetRelativeMouseMode(sdl_on_wayland ? SDL_FALSE : SDL_TRUE);
 				SDL_WarpMouseInWindow(this->sdl_window, _cursor.pos.x, _cursor.pos.y);
+			} else {
+				SDL_SetRelativeMouseMode(SDL_FALSE);
 			}
 			HandleMouseEvents();
 			break;
-- 
2.39.1

