From a9372026f9a21611bac5326de0e1a4e39e0cab49 Mon Sep 17 00:00:00 2001
From: Ghislain Antony Vaillant <ghisvail@gmail.com>
Date: Wed, 17 Apr 2019 22:01:05 +0200
Subject: [PATCH] Add soundfont path for Flatpak

---
 src/music/fluidsynth.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/music/fluidsynth.cpp b/src/music/fluidsynth.cpp
index 6baeb899b..c5b21565c 100644
--- a/src/music/fluidsynth.cpp
+++ b/src/music/fluidsynth.cpp
@@ -39,6 +39,9 @@ static const char *default_sf[] = {
 	"/usr/share/sounds/sf2/TimGM6mb.sf2",
 	"/usr/share/sounds/sf2/FluidR3_GS.sf2",
 
+	/* Flatpak */
+	"/app/share/sounds/sf2/FluidR3_GM.sf2",
+
 	NULL
 };
 
-- 
2.20.1

