From 0b026778dad96ded3e3d1a5def8bfe0465f1c0f4 Mon Sep 17 00:00:00 2001
From: Ghislain Antony Vaillant <ghisvail@gmail.com>
Date: Wed, 17 Apr 2019 22:01:05 +0200
Subject: [PATCH] Add soundfont path for Flatpak


diff --git a/src/music/fluidsynth.cpp b/src/music/fluidsynth.cpp
index 64abe0e53..04e913b27 100644
--- a/src/music/fluidsynth.cpp
+++ b/src/music/fluidsynth.cpp
@@ -39,6 +39,9 @@ static const char *default_sf[] = {
 	"/usr/share/sounds/sf2/TimGM6mb.sf2",
 	"/usr/share/sounds/sf2/FluidR3_GS.sf2",
 
+	/* Flatpak */
+	"/app/share/sounds/sf2/FluidR3_GM.sf2",
+
 	nullptr
 };
 
-- 
2.20.1

